lib/ansible/modules/system/ufw.py

```diff
@@ -309,22 +318,41 @@ def ufw_version():
                 if value in ['reloaded', 'reset']:
                     changed = True
                 else:
-                    (_, ufw_state, _) = execute([[ufw_bin], ["status"]])
-                    ufw_enabled = ufw_state.find("active") != -1
-                    if (value == 'disabled' and ufw_enabled) or (value == 'enabled' and ufw_enabled):
+                    ufw_enabled = pre_state.find("active") != -1
+                    if (value == 'disabled' and ufw_enabled) or (value == 'enabled' and not ufw_enabled):
                         changed = True
             else:
                 execute(cmd + [['-f'], [states[value]]])
 
         elif command == 'logging':
             if module.check_mode:
-                changed = True
+                extract = re.search(r'Logging: ([onf]+) \(([a-z]+)\)', pre_state)
+                if extract:
+                    current_level = extract.group(1)
+                    current_on_off_value = extract.group(2)
+                    if value != "off":
+                        if value != current_level:
+                            changed = True
+                    elif current_on_off_value != "off":
+                        changed = True
+                else:
+                    changed = True
+
             else:
                 execute(cmd + [[command], [value]])
 
         elif command == 'default':
             if module.check_mode:
-                changed = True
+                extract = re.search(r'Default: ([denyalow]+) \(incoming\), ([denyalow]+) \(outgoing\), ([denyallow]+) \(routed\)', pre_state)
+                if extract:
+                    defaults = {}
```

Comment:
I would rather say `current_values` instead of `defaults`, since these are not the default values, but the current values.

Reply:
That's refer to the firewall default rules, I'll rename to current_default_values.

Reply:
Ah. Yes, good idea; the new name is much better and won't confuse me that much ;)